index.js

```diff
@@ -13,31 +12,33 @@ export default function icomesh(order = 4) {
         f, 0, -1, f, 0, 1, -f, 0, -1, -f, 0, 1
     ]);
 
-    let triangles = new Uint16Array([
+    let triangles = Uint16Array.of(
         0, 11, 5, 0, 5, 1, 0, 1, 7, 0, 7, 10, 0, 10, 11,
         11, 10, 2, 5, 11, 4, 1, 5, 9, 7, 1, 8, 10, 7, 6,
         3, 9, 4, 3, 4, 2, 3, 2, 6, 3, 6, 8, 3, 8, 9,
         9, 8, 1, 4, 9, 5, 2, 4, 11, 6, 2, 10, 8, 6, 7
-    ]);
+    );
 
     let v = 12;
-    const midCache = order && new Map(); // midpoint vertices cache to avoid duplicating shared vertices
+    const midCache = order ? new Map() : null; // midpoint vertices cache to avoid duplicating shared vertices
 
     function addMidPoint(a, b) {
-        const key = Math.floor((a + b) * (a + b + 1) / 2) + (a < b ? a : b); // Cantor's pairing function
+        const key = (((a + b) * (a + b + 1) / 2) + Math.min(a, b)).toString(); // Cantor's pairing function
```

Comment:
Pulling your PR, the bench surprisingly showed a noticeable (~20–50%) slowdown vs `master`, and this `toString` here is the culprit. Apparently v8 in Node v12 optimized numeric keys in `Map`?

Reply:
Hmm, that's strange I got this results on `node v12.10.0`:
before:
```
Generated 9-order icosphere (2621442 vertices / 5242880 triangles) in 6,819.921ms.
Generated 10-order icosphere (10485762 vertices / 20971520 triangles) in 120,752.467ms.
```
after:
```
Generated 8-order icosphere (655362 vertices / 1310720 triangles) in 580.519ms.
Generated 9-order icosphere (2621442 vertices / 5242880 triangles) in 2,670.783ms.
```
For order less than 8 result pretty similar

Reply:
`Map`/`Set` works the best when key integer or string (which hashed to integer under the hood) and works really bad when key is double / float